subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/OperationDependenciesLookup.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling.internal.provider.runner;
+
+import org.gradle.execution.plan.ExecutionDependencies;
+import org.gradle.tooling.internal.protocol.events.InternalOperationDescriptor;
+
+import java.util.stream.Stream;
+
+@FunctionalInterface
+interface OperationDependenciesLookup {
```

Comment:
Why is this interface is implemented by two _listeners?_

Reply:
... and only operations that have been reported by listeners are relevant.

Reply:
This still feels wrong somehow. A `@`FunctionalInterface` is supposed to be something we pass to methods taking, well, _a function._ A listener on the other hand is something that is never passed around except for when we register it with the listener manager.

Reply:
You're right, forget about it being a `@FunctionalInterface`, that's a leftover from some intermediate state which I'll remove. But besides that, I pass listeners around in this area of the code all the time. Most track some data, e.g. plugin applications, and provide them to other listeners.

Reply:
Interesting. I still don't get how a listener is supposed to return values. But if this is a pattern used in the tooling API, I don't want to harass you further with this. :)

Reply:
I've removed `@FunctionalInterface` in 43ae737563d.